src/deterministic_vi/elbo_args.jl

```diff
@@ -50,78 +50,75 @@ type ElboIntermediateVariables{NumType <: Number}
     E_G2_s_hsub_vec::Vector{HessianSubmatrices{NumType}}
 
     # Expected pixel intensity and variance for a pixel from all sources.
-    E_G::SensitiveFloat{CanonicalParams, NumType}
-    var_G::SensitiveFloat{CanonicalParams, NumType}
+    E_G::SensitiveFloat{NumType}
+    var_G::SensitiveFloat{NumType}
 
     # Pre-allocated memory for the gradient and Hessian of combine functions.
     combine_grad::Vector{NumType}
     combine_hess::Matrix{NumType}
 
     # A placeholder for the log term in the ELBO.
-    elbo_log_term::SensitiveFloat{CanonicalParams, NumType}
+    elbo_log_term::SensitiveFloat{NumType}
 
     # The ELBO itself.
-    elbo::SensitiveFloat{CanonicalParams, NumType}
-
-    # If false, do not calculate hessians or derivatives.
-    calculate_derivs::Bool
-
-    # If false, do not calculate hessians.
-    calculate_hessian::Bool
+    elbo::SensitiveFloat{NumType}
 end
 
 
 """
 Args:
     - S: The total number of sources
     - num_active_sources: The number of actives sources (with deriviatives)
-    - calculate_derivs: If false, only calculate values
+    - calculate_gradient: If false, only calculate values
     - calculate_hessian: If false, only calculate gradients. Note that if
-                calculate_derivs = false, then hessians will not be
+                calculate_gradient = false, then hessians will not be
                 calculated irrespective of the value of calculate_hessian.
 """
 function ElboIntermediateVariables(NumType::DataType,
                                    S::Int,
                                    num_active_sources::Int;
-                                   calculate_derivs::Bool=true,
+                                   calculate_gradient::Bool=true,
```

Comment:
As long as you're making all these changes, I'd like to mention that I never quite understood why ```calculate_gradient``` and ```calculate_hessian``` aren't included in ```ElboArgs```.